numpy/core/memmap.py

```diff
@@ -335,3 +342,24 @@ def __getitem__(self, index):
         if type(res) is memmap and res._mmap is None:
             return res.view(type=ndarray)
         return res
+
+    def close(self):
+        """
+        Close the memmap file and reduce reference count by 1.
+        If the reference count is zero, the file is unlocked.
+        
+        It has no effect if no file is memory-mapped.
+
+        For further information, see `memmap`.
+
+        Parameters
+        ----------
+        None
+
+        See Also
+        --------
+        memmap
+
+        """
+        if self._mmap:
```

Comment:
Can you explain what the purpose is of this `if self._mmap: ...` statement?

Reply:
`self._mmap` may be None if it is not mapped or unmapped. Hence I did a check before calling `.mmap.close()`.  Hence calling the function `close()` from the same `fp` has no impact, consistent with the standard Python `file.close()`.

Reply:
``` diff
- if self._mmap:
+ if self._mmap is not None:
```
This should be an `if not None` check in that case, 
especially since `if mmap(...): ...` raises an error if the file is already closed.

Reply:
You are correct. Thanks.  I have tested the code again, it seems that numpy memmap() does not crash anymore. Need further testing.